test/unit/Container.spec.ts

```diff
@@ -128,13 +112,13 @@ describe("Container", function() {
             }
         }
 
-        function createCar (engine: Engine) {
-            return new Car(engine);
+        class CarFactory {
+            @Factory()
+            public static createCar (engine: Engine): Car {
```

Comment:
also is it okay that we can defined such "factory" anywhere? I don't think so. Just imagine you have a big codebase and you have lets say `Car` service and somewhere deep in the app I can create a `EverythingFactory` and put there logic of car creation... I think it should be explicit and we should put `@Service({ factory: carFactory }) export class Car...` to make it really clear and make flow explicit. If you don't like that your function will be inside the decorator - then don't worry - it may not be there. Users just need to create `export function carFactory(){ ... }` in a separate file and use `@Service({ factory: carFactory })` where `carFactory` is imported from that file.